registry/modules/authBasic/controllers/authBasic.ts

```diff
@@ -18,109 +18,92 @@ import {
   RefreshTokenRepository,
 } from "@/repositories/refreshToken.interface.js";
 
-interface TokensOutput { accessToken: string; refreshToken: RefreshToken };
-interface AuthOutput extends TokensOutput  {user: Omit<User, 'password'>}
+interface TokensOutput {
+  accessToken: string;
+  refreshToken: RefreshToken;
+}
+interface AuthOutput extends TokensOutput {
+  user: Omit<User, "password">;
+}
 
 interface AuthBasicController {
-  login: (props: BasicAuthSchema) => Promise<AuthOutput>
-  signup: (props: BasicAuthSchema) => Promise<AuthOutput>
-  refreshToken: (props:RefreshTokenSchema) => Promise<TokensOutput>
+  login: (props: BasicAuthSchema) => Promise<AuthOutput>;
+  signup: (props: BasicAuthSchema) => Promise<AuthOutput>;
+  refreshToken: (props: RefreshTokenSchema) => Promise<TokensOutput>;
 }
 
 export const createAuthBasicController = (
   userRepo: UserRepository,
   refreshTokenRepo: RefreshTokenRepository
 ): AuthBasicController => {
-  const generateAccessToken = (userId: number) => {
-    const signedJWT = accessTokenManager.sign({ userId: userId.toString() });
+  const generateTokens = async (userId: number, tokenFamily?: string) => {
+    const [refreshToken, accessToken] = await Promise.all([
+      generateRefreshToken(userId, tokenFamily),
+      generateAccessToken(userId),
+    ]);
+    return { accessToken, refreshToken };
+  };
 
-    return signedJWT;
+  const generateAccessToken = (userId: number) => {
+    return accessTokenManager.sign({ userId: userId.toString() });
   };
 
   const generateRefreshToken = async (userId: number, tokenFamily?: string) => {
-    const expAt = new Date(new Date().getTime() + 31 * 24 * 60 * 60000); // Expire in 31 days
-    const refreshTokenExp = expAt.toISOString();
-
-    const token = await refreshTokenRepo.createToken({
+    const expAt = new Date(Date.now() + 31 * 24 * 60 * 60 * 1000);
+    return refreshTokenRepo.createToken({
       userId,
       tokenFamily,
-      expiresAt: refreshTokenExp,
+      expiresAt: expAt.toISOString(),
     });
+  };
 
-    return token;
+  const getUserOrFail = async (username: string) => {
+    const user = await userRepo.getUser(username);
+    if (!user) throw invalidLoginCredentials();
+    return user;
   };
```

Comment:
We can pass `error: HttpError` here as well and reuse this function in the signup controller as well. Then you can throw different error based on the use case.